### ISSUE TITLE
Duplicates in produced css code

### ISSUE THREAD

**Issue by vicdup** [Evaneos/vitaminjs#384]
Hi there,

I am struggling to understand why in my css code I have duplicates of my main.global.css file.
This file is applied roughly 236 times thus making inspecting very slow and css very big...

I've read stuff about using importLoaders in css-loader config but that seems to be already the case. Do you know what could cause this issue ?

We have a vars.css file exporting variables with the @value notation
```
@value _firstColor: #163467;
@value _secondColor: #FFCE02;
@value _thirdColor: #4B4C50;
```

we import them from the other files
```
@value  _lightGrey, _deepBlue, _blue, _white, _melrose from '../../Ressources/css/vars.css';
```

We have some styles defined in our main.global.css : 
```
.title {
    font-weight: _fontWeightRegular;
    font-size: _fontSizeTitle;
    color: _deepBlue;
    margin: 0;
}
```

that later we use to compose in our component specific files :
```

.title {
    composes: title from "../../Ressources/css/main.global.css";
}
```
I don't know if any of this could be the issue ?

Any help would be really appreciated

**Comment by LucasPerso** [comment]
Hello Vicdup,

CSS Modules values are not entirely supported at the moment as [postcss-modules-values](https://github.com/css-modules/postcss-icss-values) is not part of VitaminJS webpack config. In the past, we have tried to integrate this plugin without success. It is surely something we should fix soon in the next version.

**Comment by vicdup** [comment]
Hey Lucas, thanks for your answer, however after further research we've been able to isolate the compose as the cause of this issue.

I tried removing all the imported values without success. What are your advices on compose ? How can we "extend" another class correctly ?

Thanks,
Victor

**Comment by alepee** [comment]
Hey @vicdup. As @LucasPerso suggested I would recommend you to stick with standard CSS syntax. You can use `custom-properties` with `@apply` keyword, as documented here http://cssnext.io/features/#custom-properties-set-apply.

```css
:root {
    --base: {
        color: red;
    };
}

.extended {
    @apply --base;
}
```

In your case, if `title` className is applied within your component, it will be copied once for this component, even if you use it multiple time within your parent component.

Keep in mind that *every time* you use `@apply` or any other so-called "extends" keywords you will duplicate CSS in the end result. CSS does not play with reference or any other kind of real variables for now. It will always end up with a copy/paste.

I am doing some tests and will let you know 😏

**Comment by heymath** [comment]
Without hot reload, CSS is twice inside the first <style></style> and once is in an other <style></style> too.. 😞 
So CSS is declared three times instead of only once, and in any environment (dev, prod, etc.) 😭 
Maybe related to #384, I'm not sure I understood the anwsers given

Counter example in dev env:
![cdd-duplicated-dev-env](https://user-images.githubusercontent.com/1656697/36041226-21669654-0dc8-11e8-96b8-44cdff60108f.png)

Counter example in prod env:
![cdd-duplicated-dev-prod](https://user-images.githubusercontent.com/1656697/36041209-160ca460-0dc8-11e8-949a-4b2b6eef803e.png)
